pkg/controller/host/host_controller.go

```diff
@@ -1519,6 +1520,94 @@ func (r *ReconcileHost) ReconcileResource(client *gophercloud.ServiceClient, ins
 	return err
 }
 
+// Function to obtain ceph replication factor
+func CephReplicationFactor(client *gophercloud.ServiceClient) (rep int, err error) {
+	n := 0
+	result, err := storagebackends.ListBackends(client)
+	if err == nil {
+		rep := result[0].Capabilities.Replication
+		n, err = strconv.Atoi(rep)
+		if err != nil {
+			return n, err
+		}
+	} else {
+		return n, err
+	}
+
+	log.Info("Ceph replication factor", "num", n)
+	return n, nil
+}
+
+// Function to judge the specified host is in the ceph
+// primary group or not. If host name is start with
```

Comment:
We don't have the naming convention to assign the storage nodes as "storage-0", "storage-1" etc. So this check cannot use the host name. I'll recommend to use the host id and its personality to do this check

Reply:
I removed host name to identify the primary storage node group and update the patch.
Thank you for your review.